aws/kinesis-stream/main.tf

```
@@ -0,0 +1,86 @@
+locals {
+  provisioned_mode = var.stream_mode == "PROVISIONED"
+}
+
+resource "aws_kinesis_stream" "main" {
+  name                      = var.name
+  shard_count               = local.provisioned_mode ? var.shard_count : null
```

```suggestion
  shard_count               = local.is_provisioned_mode ? var.shard_count : null
```

Corresponding suggestion here. Either accept both or reject both.